sn_client/src/register.rs

```diff
@@ -58,8 +58,7 @@ impl ClientRegister {
 
     /// Create a new public Register (Anybody can write to it) and send it so the Network.
     pub async fn create_public_online(client: Client, name: XorName, tag: u64) -> Result<Self> {
```

Comment:
better change `name: XorName, tag: u64` to `RegisterAddress` to avoid potential misusing

Reply:
`name: XorName, tag: u64` is use throughout the entire high level API, changing this here means changing our entire API. I don't have a strong opinion on this but we might want to think it through as it changes app devs interface. 
Probably out of scope for this PR.